core/src/main/kotlin/com/exactpro/th2/read/db/core/impl/DataBaseMonitorServiceImpl.kt

```diff
@@ -108,44 +113,53 @@ class DataBaseMonitorServiceImpl(
     private suspend fun poolUpdates(
         dataSourceId: DataSourceId,
         startFromLastReadRow: Boolean,
+        resetStateParameters: ResetState,
         initQueryId: QueryId?,
         initParameters: QueryParametersValues,
         useColumns: Set<String>,
         updateParameters: QueryParametersValues,
-        interval: Duration,
+        interval: Long,
         updateQueryId: QueryId,
         updateListener: UpdateListener,
         messageLoader: MessageLoader
     ) {
         val properties = mapOf(
             TH2_PULL_TASK_UPDATE_HASH_PROPERTY to hashService.calculateHash(dataSourceId, updateQueryId).toString()
         )
+        val finalParameters: MutableMap<String, Collection<String>> = hashMapOf()
+        var lastResetTime = Instant.MIN
+        var firstIteration = true
 
-        val lastRow: TableRow? = when(startFromLastReadRow) {
-            true -> messageLoader.load(dataSourceId, properties)
-            else -> null
-        } ?: initQueryId?.let { queryId ->
-            dataBaseService.executeQuery(
-                dataSourceId,
-                queryId,
-                initParameters,
-            ).lastOrNull()
-        }
+        do {
+            val resetDate = resetStateParameters.calculateNearestResetDate()
+            if (firstIteration || (resetDate != null && lastResetTime.isBefore(resetDate))) {
+                firstIteration = false
+                lastResetTime = Instant.now()
 
-        fun updateParameters(lastRow: TableRow): QueryParametersValues {
-            return useColumns.associateWith {
-                val value = lastRow.columns[it] ?: error("Missing required parameter $it from init query result $lastRow")
-                listOf(value.toString())
-            }
-        }
```

Comment:
Could you please advise what was wrong with this local function? I don't see any usage in other methods so I don't understand why should we extract it into a companion object and explicitly pass the set with user columns

Reply:
I maked the poolUpdates method shorter. It was actual on the first development stage. Now it doesn't matter, I can revert if you prefer previous implementation.

Reply:
Do the 6 lines matter when the whole method is much longer? I would prefer to keep the things that are created for a particular case near the place they are used.